docs/get_started/centos_setup.md

```diff
@@ -0,0 +1,164 @@
+# Installing MXNet on CentOS
+MXNet currently supports Python, R, Julia, and Scala. For users on CentOS, MXNet provides [a docker installation guide](http://mxnet.io/get_started/docker_setup.html). If you don't have docker environment, please install MXNet step by step.
+
+Note: If you are not familiar with CentOS or building progress, we encourage you to [install or deploy MXNet environment on Ubuntu](http://mxnet.io/get_started/ubuntu_setup.html). 
+
+## Minimum Requirements
+Make sure you have the root permission, and `yum` is properly installed.
+
+To install MXNet on CentOS, you must have the following:
+
+1. gcc, g++ (4.8 or later)
+2. python2, python-numpy, python-pip, clang
+3. graphviz, jupyter (pip or yum install)
+4. OpenBLAS
+5. CUDA for GPU
+6. opencv (do not use yum to install opencv, some shared libs may not be installed) and cmake
+
+### Basic packages
+Make sure your machine is connected to Internet. A few installations need to download (`git clone` or `wget`) some packages from Internet.
+
+```bash
+	# Install gcc-4.8/make and other development tools
+	sudo yum install -y gcc
+	sudo yum install -y gcc-c++
+	sudo yum install -y clang
+
+	# Install Python, Numpy, pip and set up tools.
+	sudo yum groupinstall -y "Development Tools"
+	sudo yum install -y python27 python27-setuptools python27-tools python-pip
+	sudo yum install -y python27-numpy 
+
+	# install graphviz, jupyter
+	sudo pip install graphviz
+	sudo pip install jupyter
+```
+### Install OpenBLAS
+Note that OpenBLAS can be replaced by other BLAS libs, e.g, Intel MKL.
+
+```bash
+	# Install OpenBLAS at /usr/local/openblas
+	git clone https://github.com/xianyi/OpenBLAS
+	cd OpenBLAS
+	make -j $(($(nproc) + 1))
+	sudo make PREFIX=/usr/local install
+	cd ..
+```
+### Install CUDA for GPU
```

Comment:
This is an Optional setup instruction. So adding a line - "Setting up CUDA is optional for MXNet. If you do not have a GPU machine, you can skip this section and proceed with installation of OpenCV"
I have seen few users getting confused on is it compulsory to set up CUDA.